components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -1260,70 +1262,81 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
           ! allow ice nucleation if < -15 C and > 5% ice supersaturation
           ! use CELL-AVERAGE values, freezing of vapor
 
-          if (t(i,k).lt.icenuct .and. supi(i,k).ge.0.05) then
+         if(.not. log_predictNc) then 
+            if ( t(i,k).lt.icenuct .and. supi(i,k).ge.0.05) then
 
-             !        dum = exp(-0.639+0.1296*100.*supi(i,k))*1000.*inv_rho(i,k)  !Meyers et al. (1992)
-             dum = 0.005*exp(0.304*(zerodegc-t(i,k)))*1000.*inv_rho(i,k)   !Cooper (1986)
-             dum = min(dum,100.e3*inv_rho(i,k))
-             N_nuc = max(0.,(dum-nitot(i,k))*odt)
+               !        dum = exp(-0.639+0.1296*100.*supi(i,k))*1000.*inv_rho(i,k)  !Meyers et al. (1992)
+               dum = 0.005*exp(0.304*(zerodegc-t(i,k)))*1000.*inv_rho(i,k)   !Cooper (1986)
+               dum = min(dum,100.e3*inv_rho(i,k))
+               N_nuc = max(0.,(dum-nitot(i,k))*odt)
 
-             if (N_nuc.ge.1.e-20) then
-                Q_nuc = max(0.,(dum-nitot(i,k))*mi0*odt)
-                qinuc = Q_nuc
-                ninuc = N_nuc
-             endif
+               if (N_nuc.ge.1.e-20) then
+                  Q_nuc = max(0.,(dum-nitot(i,k))*mi0*odt)
+                  qinuc = Q_nuc
+                  ninuc = N_nuc
+               endif
+
+            endif
+         endif 
 
-          endif
 
+         if(log_predictNc) then 
+         ! Ice nucleation predicted by aerosol scheme 
+            if  ( t(i,k) .lt. 258.15 .and. supi(i,k).ge.0.05) then
```

Comment:
Do we want to use a consistent icenuct for both log_predictNC and not cases, in which case this "if" line could be moved outside the if(logpredictnc) else lines?

Reply:
I modified the code to use a consistent icenuct this in the most recent commit.